lib/engine/game/g_18_ireland/step/track.rb

```diff
@@ -15,6 +15,45 @@ def process_lay_tile(action)
             super
             @game.clear_narrow_graph
           end
+
+          def check_track_restrictions!(entity, old_tile, new_tile)
+            return if @game.loading || !entity.operator?
+
+            connected_paths = if @game.tile_uses_broad_rules?(old_tile, new_tile)
+                                @game.graph_for_entity(entity).connected_paths(entity)
+                              else
+                                # Must update the graph now.
+                                @game.clear_narrow_graph
```

Comment:
Doesn't this need to also happen when the normal graph is cleared, or is this the only time narrow_graph is referenced?

Reply:
normal graph gets cleared as part of the lay tile a few lines before this gets called.

Reply:
The only reason I ask is because since this is in check_track_restrictions, the graph won't get cleared when loading.

Reply:
It gets cleared again anyway at the end of process_lay_tile